envoy-control-core/src/main/resources/lua/ingress_rbac_logging.lua

```diff
@@ -42,24 +51,45 @@ function first_header_value_from_list(header_list, handle)
     return ""
 end
 
+function is_allowed_client(client_name, clients_allowed_to_all_endpoints)
+    for _,h in ipairs(clients_allowed_to_all_endpoints) do
+        if client_name ~= "" and h ~= "" and client_name == h then
+            return true
+        end
+    end
+
+    return false
+end
+
 function envoy_on_response(handle)
     local rbacMetadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.rbac")
+    local lua_metadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.lua") or {}
+    local allowed_client = lua_metadata["request.info.allowed_client"] or false
 
-    if rbacMetadata == nil or rbacMetadata["shadow_engine_result"] ~= "denied" then
+    if rbacMetadata == nil then
```

Comment:
Why can't we do:

```
if rbacMetadata == nil or rbacMetadata["shadow_engine_result"] ~= "denied" or allowed_client == false then
```

?

Reply:
Because, in this case if `rbacMetadata["shadow_engine_result"] ~= "denied"` then we would just return without any log. We want to log if `rbacMetadata["shadow_engine_result"] ~= "denied" and allowed_client == true` but in one mentioned by you there is `or` operator which need only one condition to pass.

Reply:
I simplified this for you: https://github.com/allegro/envoy-control/compare/allow_client...allow_client_slonka?expand=1

```lua
function envoy_on_response(handle)
    local rbacMetadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.rbac") or {}
    local lua_metadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.lua") or {}
    local allowed_client = lua_metadata["request.info.allowed_client"] or false

    if should_log_request(rbacMetadata, allowed_client) then
        log_request(lua_metadata, handle)
    end
end

function should_log_request(rbacMetadata, allowed_client)
    local is_shadow_denied = (rbacMetadata["shadow_engine_result"] or "") == "denied"

    if (not is_shadow_denied) and allowed_client == true then
        return true
    else
        return is_shadow_denied
    end
end
```

Reply:
Thanks. Applied change :)